server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java

```diff
@@ -104,10 +105,12 @@ private void createMailbox(MailboxCreationId mailboxCreationId, MailboxCreateReq
         try {
             ensureValidMailboxName(mailboxRequest, mailboxSession);
             MailboxPath mailboxPath = getMailboxPath(mailboxRequest, creationIdsToCreatedMailboxId, mailboxSession);
-            mailboxManager.createMailbox(mailboxPath, mailboxSession);
-            subscriptionManager.subscribe(mailboxSession, mailboxPath.getName());
-            Optional<Mailbox> mailbox = mailboxFactory.fromMailboxPath(mailboxPath, mailboxSession);
+            Optional<MailboxId> mailboxId = OptionalConverter.fromGuava(mailboxManager.createMailbox(mailboxPath, mailboxSession));
+            Optional<Mailbox> mailbox;
+            mailbox = mailboxId.map(id -> mailboxFactory.fromMailboxId(id, mailboxSession))
+                .orElseGet(Optional::empty);
```

Comment:
maybe I don't get it, but http://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#flatMap-java.util.function.Function- says it exists